app/controllers/repositories_controller.rb

```diff
@@ -100,14 +103,21 @@ def branches
   end
 
   def notify_push
```

Comment:
And maybe add even more conditions than it already has? I think that's not good to code quality.